src/components/TextInput/TextInput.tsx

```diff
@@ -54,16 +56,22 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
         style,
         className,
         qa,
+        innerLabelProps,
     } = props;
     const [uncontrolledValue, setUncontrolledValue] = React.useState(defaultValue ?? '');
     const innerControlRef = React.useRef<HTMLTextAreaElement | HTMLInputElement>(null);
+    const innerLabelRef = React.useRef<HTMLSpanElement>(null);
+    const [innerLabelWidth, setInnerLabelWidth] = React.useState(0);
     const [hasVerticalScrollbar, setHasVerticalScrollbar] = React.useState(false);
 
     const isControlled = value !== undefined;
     const inputValue = isControlled ? value : uncontrolledValue;
+    const isInnerLabelVisible = !multiline && Boolean(innerLabel);
```

Comment:
We can implement label via `text-indent: <number> px`. Thus we'll support both singleline and multiline inputs

Reply:
`text-indent` could be an option, but:
- label is not workable in multiline input, as on scroll the content covers label (see attach). I think at this stage Label makes sense only for singleline inputs

if we do not use label for multiline - does it make sense to use `text-indent`?

![Screenshot 2022-12-28 at 13 02 55](https://user-images.githubusercontent.com/41027510/209796081-f26e5727-65b2-4c71-bcf8-2956f8ef0ebc.png)